### PR TITLE
Nativelib/env

### PR DESCRIPTION
**Pull request by Jens0512**
Added an `ENV` module. Also added `Map` `#keys` and `#values`. 

TODO: 
- [x] Documentation

### CODE REVIEW

File: spec/myst/env_spec.mt [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  ENV.set(TEST_ENV)
+end
+
+def random_key
+  key = ""
+  16.times {key += Random.rand(10).to_s}
+  key
+end
+
+describe("ENV[]") do
```
Comment (faultyserver) [inline spec/myst/env_spec.mt]: Just a nitpick. Can all of the `ENV` specs get put under a general `describe("ENV") do`, and then have separate, nested `describe` blocks for each method?

File: spec/myst/env_spec.mt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    assert(ENV[random_key]).is_a(Nil)
+  end
+end
+
+describe("ENV.fetch") do
+  it("is just the same as ENV[] when the key is present in the environment, if not: fetch returns provided default (which defaults to nil) instead of nil; if no default is specified, an error is raised") do
+    set_test_env
+    assert(ENV.fetch(random_key, "none")) == "none"
+    assert(ENV.fetch(SHAKE_KEY)) == SHAKE
+    assert{ENV.fetch(random_key)}.raises
```
Comment (faultyserver) [inline spec/myst/env_spec.mt]: Spaces around block bodies would make this a little more obvious that the last one is a `BlockAssertion` :)

```ruby
assert{ ENV.fetch(random_key) }.raises
```

File: spec/myst/env_spec.mt [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    assert(ENV.delete(EIN_STEIN_KEY)) == TEST_ENV[EIN_STEIN_KEY]
+    assert(ENV.delete(random_key)).is_nil
+  end
+end
+
+describe("ENV[]=") do
+  it("sets the given value assigned to given key") do
+    set_test_env
+    assert(ENV[SHAKE_KEY])     == SHAKE
+    assert(ENV[EIN_STEIN_KEY]) == EIN_STEIN
```
Comment (faultyserver) [inline spec/myst/env_spec.mt]: These are testing `[]`, not `[]=`

File: spec/myst/env_spec.mt [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+end
+
+describe("ENV.set") do
+  it("sets the environment to given map") do
+    set_test_env
+    assert(ENV.to_map) == TEST_ENV
+  end
+end
+
+ENV.set(initial_env)
```
Comment (faultyserver) [inline spec/myst/env_spec.mt]: Good catch :)

File: stdlib/prelude.mt [hunk 5]
```diff
@@ -19,3 +19,4 @@ require "./map.mt"
 require "./range.mt"
 require "./string.mt"
 require "./time.mt"
+require "./env.mt"
```
Comment (faultyserver) [inline stdlib/prelude.mt:22]: keeping these in alphabetic order would be nice.

**Review by faultyserver — COMMENTED** [review]
Nice work :) Mainly just some nitpicks on style here.

File: spec/myst/env_spec.mt (hunk 1 above)
Comment (Jens0512) [reply spec/myst/env_spec.mt]: Oh I'd love to, but I thought the current spec module didn't support that yet?

File: spec/myst/env_spec.mt (hunk 3 above)
Comment (Jens0512) [reply spec/myst/env_spec.mt]: True! I think `set_test_env` originally used `[]=`, but that changed, and this did not XD.

File: spec/myst/env_spec.mt (hunk 1 above)
Comment (faultyserver) [reply spec/myst/env_spec.mt]: you can nest them, it just doesn't format the name super nicely in the error messages.

Comment (Jens0512) [reply spec/myst/env_spec.mt]: Right that was what it was

File: spec/myst/env_spec.mt (hunk 4 above)
Comment (Jens0512) [reply spec/myst/env_spec.mt]: :smile:

File: stdlib/prelude.mt (hunk 5 above)
Comment (Jens0512) [reply stdlib/prelude.mt:22]: Fixed.

(Wondering whether this will make github understand this is taken care of...)

**Comment by Jens0512** [comment]
I messed up some git stuff (again), trying to fix it. 

EDIT: There, `--force` always does the job... :sweat_smile:

File: stdlib/prelude.mt (hunk 5 above)
Comment (faultyserver) [reply stdlib/prelude.mt:22]: doesn't look like it's been edited...

Comment (Jens0512) [reply stdlib/prelude.mt:22]: What do you mean?

Comment (faultyserver) [reply stdlib/prelude.mt:22]: The diff doesn't show that this moved, and it's still at the end on master now: https://github.com/myst-lang/myst/blob/546c2877be8b58a668bd724a936a55c2c80e0308/stdlib/prelude.mt#L22

Not a big deal, though.

**Pull request merged**